dev-libs/wlroots/metadata.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>speedjack95@gmail.com</email>
+		<name>Niccolò Scatena</name>
+	</maintainer>
+	<maintainer type="project">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<longdescription>
+		Pluggable, composable, unopinionated modules for building a Wayland
+		compositor.
+
+		wlroots implements a huge variety of Wayland compositor features and
+		implements them right. By using wlroots, you get high performance,
+		excellent hardware compatibility, broad support for many wayland
+		interfaces, and comfortable development tools.
+
+		wlroots is developed under the direction of the <pkg>dev-libs/sway</pkg> project.
+	</longdescription>
+	<use>
+		<flag name="elogind">Enable support for rootless session via elogind</flag>
+		<flag name="icccm">Use xcb-icccm (from <pkg>x11-libs/xcb-util-wm</pkg>) util library for improved Xwayland introspection</flag>
+		<flag name="rootston">Build the rootston example compositor</flag>
+		<flag name="x11-backend">Enable support for handling input/output devices through <pkg>x11-libs/libxcb</pkg></flag>
+		<flag name="xkb">Use xcb-xkb (from <pkg>x11-libs/libxcb</pkg>) util library for improved keyboard handling on the X11 backend</flag>
```

Comment:
Is it worth it to use a separate `xkb` flag? If yes, would it do anything if `x11-backend` is disabled?

Reply:
`xkb` flag does something only with `x11-backend` enabled. I've just added `xkb? ( x11-backend )` as requirement.

Reply:
The `desktop` profile is default enabling `X`. This causes `x11-libs/libxcb[xkb]` to be default enabled already, because it is required by `x11-libs/libxkbcommon[X]` (by the way, do we also need that here? does `x11-backend` work with `x11-libs/libxkbcommon[-X]`?). 

Currently, `xkb` flag does not save any dependency and is only valid as a sub-option. The question I would ask myself is: do we need the added complexity for something that will be enabled by 99% of all users? And how big will be the chance of someone enabling `x11-backend` but building with `-X` globally?

Reply:
> by the way, do we also need that here? does `x11-backend` work with `x11-libs/libxkbcommon[-X]`?

The `X` flag in `libxkbcommon` just adds `xkbcommon-x11.h` that is not used by wlroots in any way, even with `x11-backend` enabled. So it should work (I have not tested it with `libxkbcommon[-X]`, but, looking at the source, I see no reason why it shouldn't work).

> And how big will be the chance of someone enabling `x11-backend` but building with `-X` globally?

I agree with you, it's probably very low.

I removed the `xkb` flag and forced `libxcb[xkb]` if `x11-backend` is active. If someone enables `x11-backend` but disables `X` globally (or locally in `libxkbcommon`), it will pull in `libxcb[xkb]` anyway - so it should work even in this case. This makes impossible to build wlroots with the X11 backend support and *without* the XCB-XKB improved keyboard handling support (just few lines of code of wlroots are not compiled - and they are probably wanted by the 99% of the users who enable `x11-backend`).